### PR TITLE
Introduce a distilled glue layer for XEN on R-Car.

### PR DESCRIPTION
**Pull request by aanisov**
From now this become a minimal glue layer to add XEN build to a release
Renesas Yocto build.
Now XEN is taken from head of master branch.
Currently it is applicable to Renesas Yocto 2.19.0.

R-Car wiki on xenproject will be updated to adopt this layer.

### CODE REVIEW

File: meta-rcar-gen3-xen/conf/layer.conf [hunk 1]
```diff
@@ -7,4 +7,4 @@ BBFILES += "${LAYERDIR}/recipes-*/*/*.bb \
 
 BBFILE_COLLECTIONS += "rcar-gen3-xen"
 BBFILE_PATTERN_rcar-gen3-xen = "^${LAYERDIR}/"
-BBFILE_PRIORITY_rcar-gen3-xen = "6"
+BBFILE_PRIORITY_rcar-gen3-xen = "6"
```
Comment (andr2000) [inline meta-rcar-gen3-xen/conf/layer.conf]: is this change needed?

Comment (aanisov) [reply meta-rcar-gen3-xen/conf/layer.conf]: not really

File: meta-rcar-gen3-xen/recipes-kernel/linux/linux-renesas/r8a7795-salvator-x-xen.dts [hunk 2]
```diff
@@ -0,0 +1,1056 @@
...
+	compatible = "renesas,salvator-x", "renesas,r8a7795";
+
+	aliases {
+		serial0 = &scif2;
+		serial1 = &scif1;
+		ethernet0 = &avb;
+	};
+
+	chosen {
+		bootargs = "dom0_mem=752M console=dtuart dtuart=serial0 dom0_max_vcpus=4 bootscrub=0 flask_enforcing=1 loglvl=all";
```
Comment (aanisov) [inline meta-rcar-gen3-xen/recipes-kernel/linux/linux-renesas/r8a7795-salvator-x-xen.dts]: flask_enforcing, loglevel, and bootscrub are odd parameters for sample configuration

File: meta-rcar-gen3-xen/recipes-kernel/linux/linux-renesas/r8a7795-salvator-x-xen.dts [hunk 3]
```diff
@@ -0,0 +1,1056 @@
...
+		xen,dom0-bootargs = "console=hvc0 root=/dev/nfs nfsroot=192.168.1.100:/srv/dom0 ip=192.168.1.10 rw rootwait rootfstype=ext4 ignore_loglevel cma=128M";
```
Comment (aanisov) [inline meta-rcar-gen3-xen/recipes-kernel/linux/linux-renesas/r8a7795-salvator-x-xen.dts]: rootfstype=ext4 is odd for nfs boot

File: meta-rcar-gen3-xen/recipes-bsp/arm-trusted-firmware/arm-trusted-firmware_git.bbappend [hunk 4]
```diff
@@ -1,2 +1,2 @@
 # Start cores in EL2 mode
-ATFW_OPT += ' RCAR_BL33_EXECUTION_EL=BL33_EL2'
+ATFW_OPT_append = ' RCAR_BL33_EXECUTION_EL=BL33_EL2'
```
Comment (andr2000) [inline meta-rcar-gen3-xen/recipes-bsp/arm-trusted-firmware/arm-trusted-firmware_git.bbappend]: no reason for this change: both have " " in front

File: meta-rcar-gen3-xen/recipes-core/images/core-image-minimal.bbappend [hunk 5]
```diff
@@ -1,10 +1,20 @@
 #Add Xen to build
 IMAGE_INSTALL_append = " \
     xen-base \
-    xen-flask \
```
Comment (andr2000) [inline meta-rcar-gen3-xen/recipes-core/images/core-image-minimal.bbappend:4]: any reason we remove flask? wouldn't it be needed for someone in this reference build?

**Review by andr2000 — APPROVED** [review]
with minor issues,
`Reviewed-by: Oleksandr Andrushchenko <oleksandr_andrushchenko@epam.com>`
Please do not use period at the end of a patch title

File: meta-rcar-gen3-xen/recipes-core/images/core-image-minimal.bbappend (hunk 5 above)
Comment (aanisov) [reply meta-rcar-gen3-xen/recipes-core/images/core-image-minimal.bbappend:4]: Flask is not being built by default. Its build is added in our development system by CONFIG_ passed through build command line.

Comment (aanisov) [reply meta-rcar-gen3-xen/recipes-core/images/core-image-minimal.bbappend:4]: Nobody enables flask in their wiki

Comment (aanisov) [reply meta-rcar-gen3-xen/recipes-core/images/core-image-minimal.bbappend:4]: Adding flask unveils dilemma of the XEN_REL value. Xenpolicy file name is extended by current XEN version (from xenbits/master it is 4.10-unstable now). Thus with xenbits/master version progress the recipe should be changed to let yocto build not fail finding xenpolicy file.

File: meta-rcar-gen3-xen/recipes-extended/xen/xen_git.bbappend [hunk 6]
```diff
@@ -1,86 +1,52 @@
...
-    file://0017-xen-arm-Add-SMC-call-function-that-is-compatible-wit.patch \
-    file://0018-xen-arm-add-basic-support-for-OP-TEE.patch \
-    file://0019-arm-optee-add-support-for-shared-memory-across-domai.patch \
-    file://0020-arm-optee-disable-SHM-cache.patch \
-    file://0021-arm-optee-add-RPC-and-session-tracking.patch \
-    file://0022-arm64-optee-Enable-OPTEE-for-arm64.patch \
-    file://0023-arm-Do-not-check-memory-bank-contiguity.patch \
-"
-EXTRA_OEMAKE += " CONFIG_HAS_SCIF=y debug=y CONFIG_EARLY_PRINTK=salvator CONFIG_QEMU_XEN=n ARM32_SEPAR_MEM_SPLIT=y"
+EXTRA_OEMAKE += " CONFIG_HAS_SCIF=y CONFIG_QEMU_XEN=n"
```
Comment (aanisov) [inline meta-rcar-gen3-xen/recipes-extended/xen/xen_git.bbappend:11]: BTW, there were no flask build enabled here (it was a patch to XEN to enable flask build in config). But we do not apply any patches to XEN.

**Review by iartemenko — COMMENTED** [review]
No questions

**Comment by andr2000** [comment]
lgtm, rb
